src/Stratis.Bitcoin.Features.SmartContracts/Rules/CanGetSenderRule.cs

```diff
@@ -6,22 +6,28 @@
 using Stratis.Bitcoin.Consensus.Rules;
 using Stratis.Bitcoin.Features.Consensus.CoinViews;
 using Stratis.Bitcoin.Features.Consensus.Rules;
+using Stratis.Bitcoin.Features.SmartContracts.PoS;
 using Stratis.SmartContracts.Core.Util;
 
 namespace Stratis.Bitcoin.Features.SmartContracts.Rules
 {
     public class CanGetSenderRule : UtxoStoreConsensusRule
     {
         private readonly ISenderRetriever senderRetriever;
+        private readonly ISmartContractActivationProvider smartContractActivationProvider;
 
-        public CanGetSenderRule(ISenderRetriever senderRetriever)
+        public CanGetSenderRule(ISenderRetriever senderRetriever, ISmartContractActivationProvider smartContractActivationProvider = null)
```

Comment:
Is there any reason we'd ever want this to default to `null`?

Reply:
The smart contract activation provider is only used when a network does not have smart contracts activated from the start.

Reply:
Right, but that logic would still be in the activation provider. On a new network the provider implementation could just return true.

Reply:
The idea is to maintain backwards compatibility with existing unrelated networks that do not have an activation provider.  Activation providers can be added for other networks as separate PRs.

Reply:
Sounds reasonable